extensions-contrib/dropwizard-emitter/src/main/java/org/apache/druid/emitter/dropwizard/DropwizardConverter.java

```diff
@@ -0,0 +1,97 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.dropwizard;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import org.apache.druid.java.util.common.ISE;
+import org.apache.druid.java.util.common.logger.Logger;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+/**
+ *
+ */
+public class DropwizardConverter
+{
+  private static final Logger log = new Logger(DropwizardConverter.class);
+  private Map<String, DropwizardMetricSpec> metricMap;
+
+  public DropwizardConverter(ObjectMapper mapper, String dimensionMapPath)
+  {
+    metricMap = readMap(mapper, dimensionMapPath);
+  }
+
+  @Nullable
+  public DropwizardMetricSpec addFilteredUserDims(
+      String service,
+      String metric,
+      Map<String, Object> userDims,
+      Map<String, String> builder
+  )
+  {
+     /*
+        Find the metric in the map. If we cant find it try to look it up prefixed by the service name.
+        This is because some metrics are reported differently, but with the same name, from different services.
+       */
+    DropwizardMetricSpec metricSpec = null;
+    if (metricMap.containsKey(metric)) {
+      metricSpec = metricMap.get(metric);
+    } else if (metricMap.containsKey(service + "-" + metric)) {
+      metricSpec = metricMap.get(service + "-" + metric);
+    }
+    if (metricSpec != null) {
+      for (String dim : metricSpec.getDimensions()) {
+        if (userDims.containsKey(dim)) {
+          builder.put(dim, userDims.get(dim).toString());
+        }
+      }
+      return metricSpec;
+    } else {
+      return null;
+    }
+  }
+
+  private Map<String, DropwizardMetricSpec> readMap(ObjectMapper mapper, String dimensionMapPath)
+  {
+    try {
+      InputStream is;
+      if (Strings.isNullOrEmpty(dimensionMapPath)) {
+        log.info("Using default metric dimension and types");
+        is = this.getClass().getClassLoader().getResourceAsStream("defaultMetricDimensions.json");
+      } else {
+        log.info("Using metric dimensions at types at [%s]", dimensionMapPath);
+        is = new FileInputStream(new File(dimensionMapPath));
+      }
+      return mapper.readerFor(new TypeReference<Map<String, DropwizardMetricSpec>>()
+      {
+      }).readValue(is);
+    }
+    catch (IOException e) {
```

Comment:
how about closing the stream on exception ?

Reply:
done.

Reply:
Hmm, why is `is` not closed on normal exit?

Reply:
thought ObjectReader.readValue will close the stream in normal flow

Reply:
` ObjectReader.readValue(..)` doesn't mention closing the InputStream by itself, it might be safer to close it in finally block.

Reply:
done.